sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/AutoConfigurationCustomizer.java

```diff
@@ -66,4 +66,18 @@ AutoConfigurationCustomizer addSpanExporterCustomizer(
    */
   AutoConfigurationCustomizer addPropertiesSupplier(
       Supplier<Map<String, String>> propertiesSupplier);
+
+  /**
+   * Control the registration of a shutdown hook to shut down the SDK when appropriate.
+   *
+   * <p>Skipping the registration of the shutdown hook may cause unexpected behavior. This
+   * configuration is for SDK consumers that require control over the SDK lifecycle. In this case,
+   * alternatives must be provided by the SDK consumer to shut down the SDK.
+   *
+   * @param registerShutdownHook a boolean <code>true</code> will register the hook, otherwise
+   *     <code>false</code> will skip registration.
+   */
+  default AutoConfigurationCustomizer registerShutdownHook(boolean registerShutdownHook) {
```

Comment:
Is there anyway to unit test this?

Reply:
There is no public API to query the registered shutdown hooks. One way would be to keep a reference to the `Thread` and then call `removeShutdownHook`.